docs: README.md update (#1114)
canonical/k8s-snap: README.md

# Canonical Kubernetes Snap

[![End to End Tests](https://github.com/canonical/k8s-snap/actions/workflows/integration.yaml/badge.svg)](https://github.com/canonical/k8s-snap/actions/workflows/integration.yaml)
![](https://img.shields.io/badge/Kubernetes-1.32-326de6.svg)

[![Get it from the Snap Store](https://snapcraft.io/static/images/badges/en/snap-store-black.svg)](https://snapcraft.io/k8s)

**Canonical Kubernetes** is the fastest, easiest way to deploy a
fully-conformant Kubernetes cluster. Harnessing pure upstream Kubernetes, this
distribution adds the missing pieces (e.g. ingress, dns, networking) for a
zero-ops experience.

## Basic usage

Easily interact with the built-in features such as the load balancer:

```bash

> sudo k8s get load-balancer

Reviewer: I disagree with removing the install instructions.

# Canonical Kubernetes Snap

[![End to End Tests](https://github.com/canonical/k8s-snap/actions/workflows/integration.yaml/badge.svg)](https://github.com/canonical/k8s-snap/actions/workflows/integration.yaml)
[![](https://snapcraft.io/k8s/badge.svg)](https://snapcraft.io/k8s)

[![Get it from the Snap Store](https://snapcraft.io/static/images/badges/en/snap-store-black.svg)](https://snapcraft.io/k8s)

**Canonical Kubernetes** is an opinionated distribution of Kubernetes which

> includes everything needed to create and manage a scalable

Reviewer 2: ```suggestion
includes all the tools needed to create and manage a scalable
```
Everything sounds a bit vague in this case.

[Cloud Native Computing Foundation](https://www.cncf.io)-conformant cluster with

> LTS. Canonical Kubernetes builds on the main Kubernetes project by providing all

Reviewer 2: Do we have a link to a LTS page that we can add?

> the necessary pieces for a zero-ops experience, such as ingress, DNS,

Reviewer 2: ```suggestion
the necessary pieces for a zero-ops experience, such as Ingress, DNS,
```
I think we capitalize Ingress in our docs- could you double check that please?

networking, and so on. Whether you are a complete beginner to Kubernetes or a

> seasoned system admin, Canonical Kubernetes provides a way to easily deploy a

Reviewer 2: ```suggestion
seasoned system administrator, Canonical Kubernetes provides a way to easily deploy a
```
I would prefer avoiding abbreviations

cluster allowing you to focus on applications over infrastructure.

## Basic usage

Canonical Kubernetes provides a way for you to easily enable, disable, and

> configure the essential default Kubernetes features in your cluster.

Reviewer 2: When I think of basic use I would first think of a CNI and coredns. I would mention the features that are enabled by default first and then continue with this section for enabling load-balancer ...

Reviewer 2: Maybe k8s status would also be helpful in this basic usage section. Wdyt?

Author: With this proposed new format, the purpose of the basic usage section is to try highlight some of the features of the product/ get a feel for the software. I tried to choose an example that highlighted the simplicity and 'plug and play' nature of the k8s features in the snap.  

I chose load balancer because it is not something you would expect to get by default in a k8s cluster but is still very helpful. I also mention we include "ingress, DNS, networking, and so on" in the intro so I wanted to try include more variety of the features in the short README.

The k8s status is helpful but I don't think it is one of the stand out features of the snap.

For example, if you want a load balancer with L2 mode enabled, run:

```bash
sudo k8s enable load-balancer
sudo k8s set load-balancer.l2-mode=true
```

Or, if you want to disable the default local storage before implementing your
own storage solution, run:

```bash
sudo k8s disable local-storage
```

Use kubectl to interact with k8s just as you would with any other Kubernetes
cluster:

```bash
sudo k8s kubectl get pods -A
```

If you want to explore the possibilities of what you can do with Canonical
Kubernetes, be sure to check out its
[how-to guides](https://documentation.ubuntu.com/canonical-kubernetes/latest/snap/howto/)
.

## Installation

Canonical Kubernetes is available for Ubuntu 22.04 and higher. It's also
available in other Linux distributions that support
[snaps](https://snapcraft.io/).

Install the snap:

```bash
sudo snap install k8s --channel=1.32-classic/stable --classic
```

Initialize the cluster:

```bash

> sudo k8s bootstrap

Reviewer 2: Should we link the bootstrap options page?

Author: I think the following line regarding customizing your deployment would cover bootstrap options

sudo k8s kubectl get pods -A
```

If you want explore more possibilities of what you can do with Canonical
Kubernetes be sure to check out our
[how-to guides](https://documentation.ubuntu.com/canonical-kubernetes/latest/snap/howto/)
.

## Installation

> Canonical Kubernetes is compatible with Ubuntu 22.04 or later. It is also

Reviewer: Can Installation be the first section?

Author: This was a conscious decision to put basic usage first. The thought is, if someone is drawn in enough by the opening paragraph to install, they will scroll down one section. If they are not yet convinced about the product, the basic usage should highlight the perks of using the software and then hopefully they will be enticed to install.

Reviewer: Ah that makes sense!

available to other operating systems that support snap packages.

Get a single node cluster installed and intialized with just two commands:

```bash
sudo snap install k8s --channel=1.32-classic/stable --classic
sudo k8s bootstrap
```

If you would like customize the deployment, see
our

> [install guides](https://documentation.ubuntu.com/canonical-kubernetes/latest/snap/howto/install/)
> .

Reviewer: ```suggestion
[install guides](https://documentation.ubuntu.com/canonical-kubernetes/latest/snap/howto/install/).
```

## Documentation

For more information and instructions, please see the our
[official documentation](https://documentation.ubuntu.com/canonical-kubernetes/)
.

## Community and support

Do you have questions about Canonical Kubernetes? Perhaps you’d like some advice
from more experienced users or discuss how to achieve a certain goal? There are
a number of ways to get in touch:

- Using the [Kubernetes slack](http://slack.kubernetes.io/):
find us in the #canonical-kubernetes channel
- On the [Ubuntu Discourse](https://discourse.ubuntu.com/c/kubernetes/180)

Canonical Kubernetes is covered by the
[Ubuntu Code of Conduct](https://ubuntu.com/community/ethos/code-of-conduct).

## Contribute to Canonical Kubernetes

Canonical Kubernetes is proudly an open source project and we welcome and
encourage contributions to the code and documentation. If you are interested
, take a look at our

> [contributing guide](https://documentation.ubuntu.com/canonical-kubernetes/latest/snap/howto/contribute/)
> .

Reviewer: ```suggestion
[contributing guide](https://documentation.ubuntu.com/canonical-kubernetes/latest/snap/howto/contribute/).
```
